components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilder.java

```diff
@@ -102,7 +104,18 @@ public DefaultIDTokenBuilder() throws IdentityOAuth2Exception {
     public String buildIDToken(OAuthTokenReqMessageContext request, OAuth2AccessTokenRespDTO tokenRespDTO)
             throws IdentityOAuth2Exception {
 
+        //issuer read from identity.xml file as first priority. If it is not there, then server url will load from carbon.xml file configs.
         String issuer = config.getOpenIDConnectIDTokenIssuerIdentifier();
+
+        if (issuer == null || issuer.isEmpty()) {
+            try {
+                issuer = IdentityUtil.getServerURL() + CONTEXT_TOKEN_ENDPOINT ;
+            } catch (IdentityException e) {
+                log.error("Error occurred while getting serverURL " + e.getMessage());
+                throw new IdentityOAuth2Exception(e.getMessage(), e);
```

Comment:
In this case I think its ok. because original error occur in identity.core level while getting server URL and there is no effect from this method for this error.